internal/pkg/plugininstaller/kubectl/installer.go

```diff
@@ -0,0 +1,111 @@
+package kubectl
+
+import (
+	"os"
+	"time"
+
+	"github.com/cenkalti/backoff"
+
+	"github.com/devstream-io/devstream/internal/pkg/plugininstaller"
+	"github.com/devstream-io/devstream/internal/pkg/plugininstaller/util"
+	"github.com/devstream-io/devstream/pkg/util/kubectl"
+)
+
+// InstallByDownload will download file for apply
+func InstallByDownload(downloadUrl string) plugininstaller.BaseOperation {
+	processFunc := func(options plugininstaller.RawOptions) error {
+		// generate k8s config file for apply
+		configFileName, err := createKubectlFile(downloadUrl, "", options)
+		defer os.Remove(configFileName)
```

Comment:
There are too many duplicate code here, and it is easy to forget `defer os.Remove(configFileName)`.

What if `createKubectlFile` just returns a `io.Reader`, and we write a new function to transfer `io.Reader` to a `os.File` and close file in this function.

And I notice that there are many operations that transfer  `io.Reader` to `os.File` in `createKubectlFile`. Maybe just return `io.Reader`.

eg:

```
// it behaves like createKubectlFile, but return io.Reader
func createKubeIOReader(...) io.Reader, nil

func KubeApply(os.File) error{}

function ExecuteFromIOReader(reader io.Reader, kubeCmd func(filename string)) {
  tempFile := fromIoReaderToTempFile()

  kubeCmd(tempFile)   //return error if neeeded
  
  defer tempFile.Close()

}

func InstallByDownload(downloadUrl string) plugininstaller.BaseOperation {
	processFunc := func(options plugininstaller.RawOptions) error {
		// generate k8s config file for apply
		reader, err := createKubectlIOReader(downloadUrl, "", options)
		if err != nil {
			return err
		}
		// kubectl apply -f
		if err = kubectl.ExecuteFromIOReader(reader, kubectl.KubeApply); err != nil {
			return err
		}
		return nil
	}
	return processFunc
}

```


Just an idea. Maybe this is way is too complex and not recommended. You could ignore this comment or consider it in the next prs.